src/utils/MainLoggerThread.php

```diff
@@ -83,24 +106,62 @@ private function writeLogStream($logResource) : void{
 				$this->notify(); //if this was due to a sync flush, tell the caller to stop waiting
 			}
 		});
+		return true;
 	}
 
-	public function run() : void{
-		$logResource = fopen($this->logFile, "ab");
+	/** @return resource */
+	private function openLogFile(string $file, int &$size){
+		$logResource = fopen($file, "ab");
 		if(!is_resource($logResource)){
 			throw new \RuntimeException("Couldn't open log file");
 		}
+		$stat = fstat($logResource);
+		if($stat === false) throw new AssumptionFailedError("ftell() should not fail here");
+		$size = $stat['size'];
+		return $logResource;
+	}
+
+	private function compressLogFile() : void{
+		$i = 0;
+		$date = date("Y-m-d\TH.i.s");
+		do{
+			//this shouldn't be necessary, but in case the user messes with the system time for some reason ...
+			$out = $this->archiveDir . "/server.${date}_$i.log.gz";
+			$i++;
+		}while(file_exists($out));
+
+		$logFile = fopen($this->logFile, 'rb');
+		$archiveFile = gzopen($out, 'wb');
+		if($logFile === false || $archiveFile === false){
+			throw new AssumptionFailedError();
+		}
+
+		if(stream_copy_to_stream($logFile, $archiveFile) === false){
+			throw new AssumptionFailedError("Something is wrong");
```

Comment:
So we can't obtain the error message here?

Reply:
This should never fail between two open valid streams. But because of the error handler it's not possible to capture the error message anyway.

Reply:
For posterity, it's possible for this to fail if the disk ran out of space during the copy.